packages/daemon/src/daemon-node-powers.js

```diff
@@ -162,6 +166,96 @@ export const makeHttpPowers = ({ http, ws }) => {
   return harden({ servePortHttp });
 };
 
+export const makeDiskPowers = ({
+  fs,
+  path: fspath,
+}) => {
+
+  /**
+   * @param {string} path
+   */
+  const makeFileReader = path => {
+    const nodeReadStream = fs.createReadStream(path);
+    return makeNodeReader(nodeReadStream);
+  };
+
+  /**
+   * @param {string} path
+   */
+  const makeFileWriter = path => {
+    const nodeWriteStream = fs.createWriteStream(path);
+    return makeNodeWriter(nodeWriteStream);
+  };
+
+  /**
+   * @param {string} path
+   * @param {string} text
+   */
+  const writeFileText = async (path, text) => {
+    await fs.promises.writeFile(path, text);
+  };
+
+  /**
+   * @param {string} path
+   */
+  const readFileText = async path => {
+    return fs.promises.readFile(path, 'utf-8');
+  };
+
+  /**
+   * @param {string} path
+   */
+  const maybeReadFileText = async path =>
+    readFileText(path).catch(error => {
+      if (
+        error.message.startsWith('ENOENT: ') ||
+        error.message.startsWith('EISDIR: ')
+      ) {
+        return undefined;
+      }
+      throw error;
+    });
+
+  /**
+   * @param {string} path
+   */
+  const readDirectory = async path => {
+    return fs.promises.readdir(path);
+  };
+
+  /**
+   * @param {string} path
+   */
+  const makePath = async path => {
+    await fs.promises.mkdir(path, { recursive: true });
+  };
+
+  /**
+   * @param {string} path
+   */
+  const removePath = async path => {
+    return fs.promises.rm(path);
+  };
+
+  const renamePath = async (source, target) =>
+    fs.promises.rename(source, target);
+
+  const joinPath = (...components) => fspath.join(...components);
+
+  return harden({
+    makeFileReader,
+    makeFileWriter,
+    writeFileText,
+    readFileText,
```

Comment:
I suppose it’s worth noting that I chose to frame these methods with `Text` in the name as opposed to only having byte-oriented APIs at the platform level. That was certainly possible, but this gives the platform latitude to take a shortcut, like Node.js. I also did not want to have an overloaded signature for both bytes and strings in the future.